R/preparedata.R

```diff
@@ -552,11 +554,7 @@ PrepareData <- function(chart.type,
     # by converting to a matrix if necessary
     if (chart.type == "Table" && !is.null(attr(data, "statistic")) &&
         (is.null(dim(data)) || length(dim(data)) == 1))
-    {
-        tmp <- attr(data, "statistic")
-        data <- as.matrix(data)
-        attr(data, "statistic") <- tmp
-    }
+        data <- CopyAttributes(as.matrix(data), data)
```

Comment:
This operation also makes me a bit nervous. A QTable that doesn't have 2 dimensions (1d array), could be upgraded to a 2d matrix, all its attribtues are copied from the 1d representation and things could break?

Reply:
I've played around with it a bit and so far I haven't noticed any problems because switching from the 1d array to a 2d matrix doesn't involve and transposing (i.e. they are all column vectors)
But if you think its too risky I can undo this, it just means the spans for this case is always dropped?